cpp/src/merge/merge.cu

```diff
@@ -457,8 +457,17 @@ table_ptr_type merge(std::vector<table_view> const& tables_to_merge,
     auto const left_table = top_and_pop(merge_queue);
     // Deallocated at the end of the block
     auto const right_table = top_and_pop(merge_queue);
-    auto merged_table =
-      merge(left_table.view, right_table.view, key_cols, column_order, null_precedence, mr, stream);
+
+    // Only use mr for the output table
+    auto const& new_tbl_rm = merge_queue.empty() ? mr : rmm::mr::get_default_resource();
+    auto merged_table      = merge(left_table.view,
+                              right_table.view,
+                              key_cols,
+                              column_order,
+                              null_precedence,
+                              new_tbl_rm,
+                              stream);
```

Comment:
Interesting - this is clearly a `clang-format` bug - might be fixed in a clang version > 8. If should be be aligning the arguments **after** it aligns the equal sign (instead of the reverse which it is clearly doing). It should be:
```cpp
    auto const& new_tbl_rm = merge_queue.empty() ? mr : rmm::mr::get_default_resource();
    auto merged_table      = merge(left_table.view,
                                   right_table.view,
                                   key_cols,
                                   column_order,
                                   null_precedence,
                                   new_tbl_rm,
                                   stream);
```